WooCommerce/Classes/ViewRelated/Blaze/BudgetSetting/BlazeBudgetSettingViewModel.swift

```diff
@@ -75,14 +77,21 @@ final class BlazeBudgetSettingViewModel: ObservableObject {
         self.timeZone = timeZone
         self.targetOptions = targetOptions
         self.stores = stores
+        self.analytics = analytics
         self.completionHandler = onCompletion
 
         observeSettings()
     }
 
+    func didTapApplyDuration() {
+        analytics.track(event: .Blaze.Budget.changedDuration(Int(dayCount)))
+    }
+
     func confirmSettings() {
-        // TODO: track confirmation
-        completionHandler(dailyAmount, Int(dayCount), startDate)
+        let days = Int(dayCount)
+        analytics.track(event: .Blaze.Budget.updateTapped(duration: days,
+                                                          totalBudget: Double(days) * dailyAmount))
```

Comment:
Nit: we should reuse `calculateTotalBudget` to avoid any incorrect calculation.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/11834/commits/82bf67c84a11f750ca309b6fe5537a9c7868882d